File: php/features.php
<?php
declare(strict_types=1);

// LmWhatsapp SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class LmWhatsappFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new LmWhatsappBaseFeature();
            case "test":
                return new LmWhatsappTestFeature();
            default:
                return new LmWhatsappBaseFeature();
        }
    }
}

File: py/features.py
# LmWhatsapp SDK feature factory

from feature.base_feature import LmWhatsappBaseFeature
from feature.test_feature import LmWhatsappTestFeature


def _make_feature(name):
    features = {
        "base": lambda: LmWhatsappBaseFeature(),
        "test": lambda: LmWhatsappTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
